pkg/asset/installconfig/powervs/validation_test.go

```diff
@@ -126,3 +129,85 @@ func TestValidate(t *testing.T) {
 		})
 	}
 }
+
+func TestValidatePreExistingPublicDNS(t *testing.T) {
+	cases := []struct {
+		name     string
+		edits    editFunctions
+		errorMsg string
+	}{
+		{
+			name:     "no pre-existing DNS records",
+			errorMsg: "",
+		},
+		{
+			name:     "pre-existing DNS records",
+			errorMsg: `^\[baseDomain\: Duplicate value\: \"record api\.valid-cluster-name\.valid\.base\.domain already exists in CIS zone \(valid-zone-id\) and might be in use by another cluster, please remove it to continue\", baseDomain\: Duplicate value\: \"record api-int\.valid-cluster-name\.valid\.base\.domain already exists in CIS zone \(valid-zone-id\) and might be in use by another cluster, please remove it to continue\"\]$`,
+		},
+		{
+			name:     "cannot get zone ID",
+			errorMsg: `^baseDomain: Internal error$`,
+		},
+		{
+			name:     "cannot get DNS records",
+			errorMsg: `^baseDomain: Internal error$`,
+		},
+	}
+	setMockEnvVars()
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	powervsClient := mock.NewMockAPI(mockCtrl)
+	metadata := mock.NewMockMetadataAPI(mockCtrl)
+
+	dnsRecordNames := [...]string{fmt.Sprintf("api.%s.%s", validClusterName, validBaseDomain), fmt.Sprintf("api-int.%s.%s", validClusterName, validBaseDomain)}
+
+	//metadata := powervs.NewMetadata(validBaseDomain)
+	//metadata.SetCISInstanceCRN(validCISInstanceCRN)
+
+	// Mocks: no pre-existing DNS records
+	metadata.EXPECT().CISInstanceCRN(gomock.Any()).Return(validCISInstanceCRN, nil)
+	powervsClient.EXPECT().GetDNSZoneIDByName(gomock.Any(), validBaseDomain).Return(validDNSZoneID, nil)
+	for _, dnsRecordName := range dnsRecordNames {
+		powervsClient.EXPECT().GetDNSRecordsByName(gomock.Any(), validCISInstanceCRN, validDNSZoneID, dnsRecordName).Return(noDNSRecordsResponse, nil)
+	}
+
+	// Mocks: pre-existing DNS records
+	metadata.EXPECT().CISInstanceCRN(gomock.Any()).Return(validCISInstanceCRN, nil)
+	powervsClient.EXPECT().GetDNSZoneIDByName(gomock.Any(), validBaseDomain).Return(validDNSZoneID, nil)
+	for _, dnsRecordName := range dnsRecordNames {
+		powervsClient.EXPECT().GetDNSRecordsByName(gomock.Any(), validCISInstanceCRN, validDNSZoneID, dnsRecordName).Return(existingDNSRecordsResponse, nil)
+	}
+
+	// Mocks: cannot get zone ID
+	metadata.EXPECT().CISInstanceCRN(gomock.Any()).Return(validCISInstanceCRN, nil)
+	powervsClient.EXPECT().GetDNSZoneIDByName(gomock.Any(), validBaseDomain).Return("", fmt.Errorf(""))
+
+	// Mocks: cannot get DNS records
+	metadata.EXPECT().CISInstanceCRN(gomock.Any()).Return(validCISInstanceCRN, nil)
+	powervsClient.EXPECT().GetDNSZoneIDByName(gomock.Any(), validBaseDomain).Return(validDNSZoneID, nil)
+	for _, dnsRecordName := range dnsRecordNames {
+		powervsClient.EXPECT().GetDNSRecordsByName(gomock.Any(), validCISInstanceCRN, validDNSZoneID, dnsRecordName).Return(nil, fmt.Errorf(""))
```

Comment:
It is a bit confusing to use the same dnsRecordNames in the tests for mocking results when you cannot find the DNS records as it is when you can find the DNS Records. Suggest making a list of records that cannot be found.